src/sentry/models/organization.py

```diff
@@ -308,12 +357,28 @@ def default_owner_id(self):
         return self._default_owner_id
 
     def has_single_owner(self):
-        from sentry.models import OrganizationMember
-
-        count = OrganizationMember.objects.filter(
-            organization=self, role=roles.get_top_dog().id, user__isnull=False, user__is_active=True
-        )[:2].count()
-        return count == 1
+        owners = list(
+            self.get_members_with_org_role(roles.get_top_dog().id).values_list("id", flat=True)
+        )
+        return len(owners[:2]) == 1
+
+    def get_members_with_org_role(self, role: str):
```

Comment:
Do you need to specify `with_org_role`?

or can this just be `get_members_with_role`?

Reply:
role could mean org role or team role so i'd want to be specific here